elasticdl/python/pserver/embedding_table.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+import tensorflow as tf
+
+
+class EmbeddingTable(object):
+    def __init__(self, name, dim=None, initializer=None):
+        self.name = name
+        self.dim = dim
+        self.initializer = initializer
+        self.embedding_vectors = {}
+
+    def get(self, indices):
+        if len(indices) == 0:
+            return None
+        values = []
+        for i in indices:
+            if i not in self.embedding_vectors:
+                init = tf.keras.initializers.get(self.initializer)
+                value = init(shape=self.dim).numpy()
+                self.embedding_vectors[i] = value
```

Comment:
If we are using synchronous SGD, we may need to add a lock when we write back to `embedding_vectors`. 

It is a little complex. Maybe leave a ToDo here and add this lock when we are implementing parameters updating process?

Reply:
Done, add a TODO here.